### PR TITLE
v5: Update Badge classes

### PR DESCRIPTION
**Pull request by thakkaryash94**
[ x ] Change .badge-variant class to .bg-variant
[ x ] Change .badge-pill class to .rounded-pill

Link: https://github.com/react-bootstrap/react-bootstrap/projects/2#card-42008293

### CODE REVIEW

File: src/Badge.tsx [hunk 1]
```diff
@@ -29,7 +30,12 @@ const propTypes = {
    * some additional horizontal padding
    */
   pill: PropTypes.bool.isRequired,
-
+  /**
+   * Sets card text color
```
Comment (kyletsang) [inline src/Badge.tsx:34]: card -> badge

Comment (thakkaryash94) [reply src/Badge.tsx]: My mistake. I was learning from card implementation. Resolved. Please check now.

File: src/Badge.tsx [hunk 2]
```diff
@@ -58,8 +65,9 @@ const Badge: Badge = React.forwardRef(
         className={classNames(
           className,
           prefix,
-          pill && `${prefix}-pill`,
-          variant && `${prefix}-${variant}`,
+          pill && `rounded-pill`,
+          text && `text-${text}`,
+          variant && `bg-${variant}`,
```
[the remark this answers is not in the document]
Comment (thakkaryash94) [reply src/Badge.tsx:70]: I was thinking the same. We are using prefix variable but I was not sure.

[the remark this answers is not in the document]
Comment (jquense) [reply src/Badge.tsx:70]: yeah basically remove variant and add "bg"

[the remark this answers is not in the document]
Comment (thakkaryash94) [reply src/Badge.tsx]: Pushed the changes. please check.

File: src/Badge.tsx [hunk 3]
```diff
@@ -18,18 +19,24 @@ const propTypes = {
   bsPrefix: PropTypes.string,
 
   /**
-   * The visual style of the badge
+   * A convenience prop for adding `bg-*` utility classes since they are so commonly used here.
+   * `light` and `dark` are common choices but any `bg-*` class is supported, including any custom ones you might define.
    *
-   * @type {('primary'|'secondary'|'success'|'danger'|'warning'|'info'|'light'|'dark')}
+   * Pairs nicely with the `variant` prop.
```
Comment (jquense) [inline src/Badge.tsx:25]: the comment is wrong here

Comment (thakkaryash94) [reply src/Badge.tsx]: Thanks, fixed now.

File: src/Badge.tsx [hunk 4]
```diff
@@ -18,18 +19,22 @@ const propTypes = {
   bsPrefix: PropTypes.string,
 
   /**
-   * The visual style of the badge
-   *
-   * @type {('primary'|'secondary'|'success'|'danger'|'warning'|'info'|'light'|'dark')}
+   * A convenience prop for adding `bg-*` utility classes since they are so commonly used here.
+   * `light` and `dark` are common choices but any `bg-*` class is supported, including any custom ones you might define.
```
Comment (jquense) [inline src/Badge.tsx:23]: this is still specific to the navbar calling out 'light' and 'dark', also still want the type into from the old comment as it it a bit more specific and still relevant

File: src/Badge.tsx [hunk 5]
```diff
@@ -4,11 +4,12 @@ import React from 'react';
 
 import { useBootstrapPrefix } from './ThemeProvider';
 import { BsPrefixProps, BsPrefixRefForwardingComponent } from './helpers';
-import { Variant } from './types';
+import { Color } from './types';
 
 export interface BadgeProps extends BsPrefixProps {
-  variant?: Variant;
+  bg?: string;
```
Comment (jquense) [inline src/Badge.tsx:10]: I think Variant still works here

Comment (thakkaryash94) [reply src/Badge.tsx]: Done. please check now.

File: src/Badge.tsx (hunk 4 above)
Comment (thakkaryash94) [reply src/Badge.tsx]: Done. please check now.

File: src/Badge.tsx [hunk 6]
```diff
@@ -19,17 +20,23 @@ const propTypes = {
 
   /**
    * The visual style of the badge
+   * A convenience prop for adding `bg-*` utility classes since they are so commonly used here.
```
Comment (kyletsang) [inline src/Badge.tsx:23]: You can delete this comment.  The bg is a core prop required for background coloring.

**Comment by bpas247** [comment]
I’m thinking we’ll probably need to squash and merge this PR once we get to that point, since the commit history got a bit messed up here

**Comment by thakkaryash94** [comment]
Yes, actually, I forgot to create a new branch.

**Comment by bpas247** [comment]
I think we're going to have to rebase this with the latest commit in the `bs5-dev` and drop the commits after 32b822edf08994a2a5e86f4e406bee52e7bc5986, seems like an issue happened with the last time this was rebased that's causing commits that are already in the upstream `bs5-dev` branch to show up as new diffs (and there are merge conflicts).

**Comment by kyletsang** [comment]
Hi @thakkaryash94, can you rebase this?

**Comment by thakkaryash94** [comment]
@bpas247 @kyletsang I am not exactly sure how to do it. I tried what @bpas247 said, can you please check. I can not see any conflicts anymore. Is this what I suppose to want.

**Comment by kyletsang** [comment]
It's still showing the diffs of all the other commits. If rebasing isn't working, I guess you could always rebranch and add the changes back since it should be very minimal.

**Comment by thakkaryash94** [comment]
I am closing the pull request and creating new one by branching out from bs5-dev.

**Pull request closed without merging**